### DISCUSSION TITLE
插入2000条数据后，按照id删除数据，query不到数据了，但是实体数量仍然是2000

### DISCUSSION THREAD

**Discussion by Qyijiu** [milvus-io/milvus#16086] [Q&A and General discussion] [answered]
![image](https://user-images.githubusercontent.com/90362695/158758247-4d3259ce-7ec3-48a1-a568-813c90c57730.png)
![image](https://user-images.githubusercontent.com/90362695/158758271-3e9b8700-96ec-4c23-9f22-3d414be237b1.png)

**Comment by yhmo** [comment]
This is a technical limitation, the collection.num_entities cannot count in deleted items.
We are building a solution for user to get accurate row count by query().

**Reply by Qyijiu** [reply]
This command has limitations, so how do I view the amount of data in the database

**Comment by yhmo** [comment, marked as answer]
A workaround is: use query() to query all the id, for example: `query(expr="id >=0")`, the count of the returned id array is the actual row count.
